components/sup/src/manager/mod.rs

```diff
@@ -1015,7 +1017,11 @@ where
             .as_ref(),
     ) {
         Ok(mut file) => {
-            match write!(&mut file, "{}", process::current_pid()) {
+            let pid = match env::var(launcher_protocol::LAUNCHER_PID_ENV) {
```

Comment:
That was exactly what I was thinking through--is there a release order to this? (I don't think it matters a lot, as long as sup/studio go together)